conan/tools/microsoft/msbuilddeps.py

```diff
@@ -65,20 +64,40 @@ class MSBuildDeps(Generator):
         </Project>
         """)
 
-    @property
-    def filename(self):
-        return None
-
-    @ staticmethod
-    def _name_condition(settings):
-        props = [("Configuration", settings.build_type),
-                 # FIXME: This probably requires mapping ARM architectures
-                 ("Platform", {'x86': 'Win32',
-                               'x86_64': 'x64'}.get(settings.get_safe("arch")))]
-
+    def __init__(self, conanfile):
+        self._conanfile = conanfile
+        self.configuration = conanfile.settings.build_type
+        self.platform = {'x86': 'Win32',
+                         'x86_64': 'x64'}.get(str(conanfile.settings.arch))
+        # TODO: this is ugly, improve this
+        self.output_path = os.getcwd()
```

Comment:
Not needed here. We are already assigning a value before calling `generate()` **with a different value** (the class is instantiated before the `chdir` call): 

```python
                    generator = generator_class(conanfile)
                    output.highlight("Generator '{}' calling 'generate()'".format(generator_name))
                    generator.output_path = path
                    with chdir(path):
                        generator.generate()
```

Reply:
But this is needed if you are explicitly instantiating the ``MSBuild()`` in the ``generate()`` method.
This is the reason it has a ``TODO``, I plan to put some time to put a nicer proposal regarding the "layout" or expected location of generated files, for implicit and explicit generators and both for cache and local flows.

Reply:
I never think about the generators like classes that the user will instantiate in the recipe 😓 

Ok, then it is needed, ugly, but needed 👍

Reply:
> I never think about the generators like classes that the user will instantiate in the recipe

But this is exactly the use case one of the test is implementing:

```python
from conans import ConanFile
            from conan.tools.microsoft import MSBuildDeps
            class Pkg(ConanFile):
                settings = "os", "compiler", "arch", "build_type"
                requires = "pkg/1.0"
                def generate(self):
                    ms = MSBuildDeps(self)
                    ms.configuration = "My"+str(self.settings.build_type)
                    ms.platform = "My"+str(self.settings.arch)
                    ms.generate()
```
The goal is to support user custom configurations in Visual Studio/MSBuild, which need to be parameterized by the users, and the best way is to be able to instantiate them directly in the ``generate()`` method. This is also how user provided custom generators (implemented in ``python_requires``) will need to be used too.

Reply:
I know, but this is something new, I've never used it in my recipes...